### DISCUSSION TITLE
Why ELF format is considered harmful for the speed of Linker?

### DISCUSSION THREAD

**Discussion by cireu** [rui314/mold#959] [Q&A] [answered]
I see the https://github.com/rui314/mold/blob/main/docs/design.md and @rui314 left a note here 

> Sometimes, the ELF file format itself seems to be a limiting factor in improving the linker's performance.

I'm not so familar with the linker technology, but I'm quite curios of what makes ELF bad for linker's perfomance. 

Answers will be appreciated.

**Comment by rui314** [comment, marked as answer]
Since I proved otherwise, it is not a valid concern.
